pkg/template/event.go

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright 2019 The Tekton Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package template
+
+import (
+	"regexp"
+	"strings"
+
+	pipelinev1 "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	"github.com/tidwall/gjson"
+	"golang.org/x/xerrors"
+)
+
+// eventPathVarRegex determines valid event path variables
+var eventPathVarRegex = regexp.MustCompile(`\$\(event(.[0-9A-Za-z_-]+)*\)`)
+
+// getEventPathFromVar returns the event path given an event path variable
+// $(event.my.path) -> my.path
+func getEventPathFromVar(eventPathVar string) string {
+	// Assume eventPathVar matches the eventPathVarRegex
+	if eventPathVar == "$(event)" {
+		return ""
+	}
+	return strings.TrimSuffix(strings.TrimPrefix(eventPathVar, "$(event."), ")")
+}
+
+// ApplyEventToParams returns the params with each event path variable replaced
+//  with the appropriate data from the event.
+// ApplyEventToParams returns an error when the event path variable is not found
+// in the event.
```

Comment:
total nitpick: im not used to seeing the name of the function mentioned twice like this, with the error behavior described on a new line - if this is common in other projects im all for it but when i saw this first I thought that the first docstring lines were accidentally duplicated.

Reply:
Not a convention, and just had the same reaction when I glanced at the code snippet here 😆 I'll update it 👍